File: doraemon-frontend/src/app/criaturas/criatura.ts
// Modelo principal - espeja la clase Criatura del backend
export interface Criatura {
  id: number;
  nombre: string;
  tipo: string;
  hp: number;
  atk: number;
  def: number;
}

// DTO para crear una nueva criatura
export interface CrearCriaturaRequest {
  nombre: string;
  tipo: string;
  hp: number;
  atk: number;
  def: number;
}

// DTO para simular un ataque
export interface AtaqueRequest {
  atacanteId: number;
  objetivoId: number;
}

// Respuesta del endpoint atacar
export interface AtaqueResultado {
  mensaje: string;
  atacante: string;
  objetivo: string;
  danio: number;
  hpRestante: number;
  derrotado: boolean;
}

// Tipos validos del juego
export const TIPOS_VALIDOS = ['Fuego', 'Agua', 'Planta', 'Roca', 'Electrico'] as const;

File: doraemon-backend/Models/Criatura.cs
namespace doraemon_backend.Models;

// =========================================================
// MODELO: Criatura
// Representa los datos de cada criatura del juego.
// =========================================================

public class Criatura
{
    public int Id { get; set; }
    public string Nombre { get; set; } = string.Empty;
    public string Tipo { get; set; } = string.Empty;  // Fuego, Agua, Planta, Roca, Electrico
    public int Hp { get; set; }
    public int Atk { get; set; }
    public int Def { get; set; }
}

// =========================================================
// MODELO: Solicitud para crear una criatura (Request DTO)
// =========================================================

public class CrearCriaturaRequest
{
    public string Nombre { get; set; } = string.Empty;
    public string Tipo { get; set; } = string.Empty;
    public int Hp { get; set; }
    public int Atk { get; set; }
    public int Def { get; set; }
}

// =========================================================
// MODELO: Solicitud de ataque entre dos criaturas
// =========================================================

public class AtaqueRequest
{
    public int AtacanteId { get; set; }
    public int ObjetivoId { get; set; }
}

// =========================================================
// MODELO: Resultado del ataque (Response DTO)
// =========================================================

public class AtaqueResultado
{
    public string Mensaje { get; set; } = string.Empty;
    public string Atacante { get; set; } = string.Empty;
    public string Objetivo { get; set; } = string.Empty;
    public int Danio { get; set; }
    public int HpRestante { get; set; }
    public bool Derrotado { get; set; }
}
